File: node/config/README.js
// 환경별로 설정파일을 따로 쓰기
const env = process.env.NODE_ENV || 'development';
const configManager = new ConfigManager(`./config.${env}.ini`);


// 설정변경을 감지하여 리로드
fs.watch(configPath, (eventType, filename) => {
  if (eventType === 'change') {
    configManager.reload();
    console.log('설정 파일이 변경되어 리로드되었습니다.');
  }
});



// TypeScript
interface AppConfig {
  database: {
    host: string;
    user: string;
    password: string;
    // ...
  };
  server: {
    port: number;
    // ...
  };
}

const config = ini.parse(fs.readFileSync('./config.ini', 'utf-8')) as AppConfig;

File: python/config/README.py
# 개발/운영 환경별 설정 관리
env = os.getenv('APP_ENV', 'development')
config = ConfigManager(f'config.{env}.ini')

# 설정 검증
class ValidatedConfigManager(ConfigManager):
    REQUIRED_KEYS = {
        'database': ['host', 'user', 'password'],
        'server': ['port']
    }

    def _load_config(self):
        super()._load_config()
        self._validate_config()

    def _validate_config(self):
        for section, keys in self.REQUIRED_KEYS.items():
            if not self.config.has_section(section):
                raise ValueError(f"필수 섹션 '{section}'가 없습니다")
            for key in keys:
                if not self.config.has_option(section, key):
                    raise ValueError(f"섹션 '{section}'에 필수 키 '{key}'가 없습니다")


